File: JavaToKotlin/app/src/main/java/io/dustin/apps/board/api/usecase/community/comment/ModifyCommentUseCase.java
package io.dustin.apps.board.api.usecase.community.comment;

import io.dustin.apps.board.domain.community.comment.service.ReadCommentService;
import io.dustin.apps.board.domain.community.comment.service.WriteCommentService;
import io.dustin.apps.board.domain.community.comment.model.Comment;
import io.dustin.apps.board.domain.community.comment.model.dto.CommentDto;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Service;
import org.springframework.web.server.ResponseStatusException;

import java.security.Principal;

@Service
@RequiredArgsConstructor
public class ModifyCommentUseCase {

    private final ReadCommentService readCommentService;
    private final WriteCommentService writeCommentService;

    public CommentDto execute(Long userId, Long commentId, String content) {
        Comment comment = readCommentService.findById(commentId);
        if (!comment.getUserId().equals(userId)) {
            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "수정권한이 없습니다.");
        }
        writeCommentService.updateContent(comment, content);
        return CommentDto.from(comment);
    }







}

File: apps/src/main/kotlin/io/dustin/apps/board/api/usecase/community/comment/ModifyCommentUseCase.kt
package io.dustin.apps.board.api.usecase.community.comment

import io.dustin.apps.board.api.community.request.command.CommentModifyCommand
import io.dustin.apps.board.domain.community.comment.model.Comment
import io.dustin.apps.board.domain.community.comment.model.dto.CommentDto
import io.dustin.apps.board.domain.community.comment.service.ReadCommentService
import io.dustin.apps.board.domain.community.comment.service.WriteCommentService
import io.dustin.apps.common.model.response.ResultResponse
import org.springframework.http.HttpStatus
import org.springframework.stereotype.Service
import org.springframework.web.server.ResponseStatusException

@Service
class ModifyCommentUseCase (

    private val readCommentService: ReadCommentService,
    private val writeCommentService: WriteCommentService,

    ) {

    fun execute(command: CommentModifyCommand): ResultResponse<CommentDto> {
        val comment: Comment = readCommentService.findByIdOrThrow(command.commentId)
        writeCommentService.updateContent(comment, command.content)
        val result = CommentDto.from(comment)
        return ResultResponse.of(result)
    }
}
